demos/dotnet/dotnet/NugetAction.cs

```diff
@@ -0,0 +1,188 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.Net.Http;
+using System.Text;
+
+namespace dotnet
+{
+    internal static class NugetAction
+    {
+        public static bool GetNugetAndRestore(ProjectProperties properties, Log log)
+        {
+            var nugetFile = Path.Combine(properties.ToolsDirectory, "nuget.exe");
+            var success = true;
+            if (!File.Exists(nugetFile))
+            {
+                success = DownloadNugetAction(properties, nugetFile);
+            }
+
+            var projectJsonFile = Path.Combine(properties.ProjectDirectory, "project.json");
```

Comment:
To implement it the way you are suggesting would require a bit more work. Can we add that as a separate issue?